### PR TITLE
Fixed LinkedList:

### PR DESCRIPTION
**Pull request by ysn551**
### Checklist

- [*] I've looked at the contribution guidelines.
- [*] This pull request is complete and ready for review.
* I am confirming that this pull request can merge [#583 Linked list conformance to collection protocol](https://github.com/raywenderlich/swift-algorithm-club/pull/583) and pass the test.

### Description

* Modified node(atIndex:) method not to return nil, and code refactoring.
* Deleted nodesBeforeAndAfter(index:) method, and modified insert methods.
* Modified readme file.

### CODE REVIEW

**Comment by hollance** [comment]
How about 'guard var node = head { ... }'. This saves having to write 'var node = head' below the guard statement.

**Comment by ysn551** [comment]
Thanks for your suggestion.
I adopted it.
https://github.com/raywenderlich/swift-algorithm-club/pull/662/commits/dc810d2a08c7f2d619a43626b1e9883e897f6f16

**Comment by ysn551** [comment]
I apologize for having sent a lot of commits after pull request.
I would be grateful if you would review this pull request.

File: Linked List/README.markdown [hunk 1]
```diff
@@ -218,37 +218,43 @@ It loops through the list in the same manner but this time increments a counter
...
 ```swift
-  public func nodeAt(_ index: Int) -> Node? {
-    if index >= 0 {
-      var node = head
-      var i = index
-      while node != nil {
-        if i == 0 { return node }
-        i -= 1
-        node = node!.next
+  public func node(atIndex index: Int) -> Node {
```
Comment (kelvinlauKL) [inline Linked List/README.markdown]: `node(at index: Int)` would read a bit more like the other collection apis.

Comment (ysn551) [reply Linked List/README.markdown]: Thank you for your review.
I renamed `atIndex`  to `at` by the following methods:
* node(atIndex:)
* insert(_:, atIndex:)
* remove(atIndex:)

https://github.com/raywenderlich/swift-algorithm-club/pull/662/commits/bb21f61b53dc08af0e5d5081951c683d0f57a979

**Comment by kelvinlauKL** [comment]
Thank you @ysn551!

**Pull request merged**